### ISSUE TITLE
Distribute binary through composer

### ISSUE THREAD

**Issue by grachevko** [spiral/roadrunner#71]
What about include build binary to composer install? 
To update need to do two thing: update composer and update binary. This is not convenient.

**Comment by Alex-Bond** [comment]
@grachevko there 2 things that comes to mind:
1. Size of the package (we have to include 4 dists in this case)
2. A lot of people using a custom build or RR

I think it's not a really good idea to do that. But it's only my opinion.

**Comment by grachevko** [comment]
@Alex-Bond it depends on how to solve that. 
The goal is remove need to duplicate version of rr in composer and binary downloader.
For example this goal can be reached by downloader script, which i can add to composer.json in `post-install-cmd`. Downloader will depend on installed rr version from composer.

If you don't need to download default binary, just not add call downloader in composer.json.
I think it's quite enough.

But It will not be superfluous to investigate how other libs solve that.

**Comment by grachevko** [comment]
@wolfy-j yeap, that is exactly what i mean.

**Comment by wolfy-j** [comment]
You can now build RR locally with or without additional services. See https://github.com/spiral/roadrunner/wiki/Quick-Builds

**Comment by Alex-Bond** [comment]
@wolfy-j how about removing Docker dep? Building RR inside Docker inside Docker (2 times) not so cool idea :)

**Comment by wolfy-j** [comment]
I do not expect it to be build in Docker but outside. This approach gives us flexibility to build any RR version based on custom specs. I'm still open to implement `rr-download` but without Symfony/Console dependency :)

**Comment by wolfy-j** [comment]
You can also compile it outside of your container, thought in this case I might need to add -os flag as currently it detect the OS on host machine.

**Comment by Alex-Bond** [comment]
@wolfy-j how about adding the flag to build in local env? Like qbuild -local.
Imagine the situation - you building RR using Jenkins in k8s. In this case, you running agent inside pod already. Running pod inside pos is not a good idea.

**Comment by wolfy-j** [comment]
It is actually intended to build in local env, this is the approach my DevOps requested (specifically to use with Jenkins). You expect to build (or use already build version) rr locally and then pack it with the rest of your applications before deployment.

**Comment by Alex-Bond** [comment]
Thats not a really good idea to have binaries inside git.

**Comment by wolfy-j** [comment]
There is no binaries in git.

**Comment by Alex-Bond** [comment]
If you compile it locally, how you will pass it to jenkins? Jenkins running whole build process inside docker.

**Comment by wolfy-j** [comment]
I'm not fully aware of the details where Jenkins is building Docker images, but you convinced me that this approach might not work for everyone. I'll try to come up with alternative.

**Comment by Alex-Bond** [comment]
@wolfy-j while preparing Sentry package i found a pretty interesting thing - how we will sen dependancy? Like what if my module depends on some internal or 3d party module and must be called only after deps injected?

**Comment by stefanos82** [comment]
I was about to open a new ticket to ask a question about the binary file for `rr`.

Shouldn't composer execute `vendor/spiral/roadrunner/build.sh` right after the package was installed? 

The only problem I have with this option is that I have to symlink the generated file to `vendor/bin/` in order to use it as I want to.

Has anyone else figure out a better way?

**Comment by stefanos82** [comment]
OK, now I'm really confused.

If that's the case, then what's the use of `composer req spiral/roadrunner symfony/psr-http-message-bridge` as described in https://github.com/spiral/roadrunner/wiki/Laravel-Framework?

If I go to `vendor` directory I can clearly see `spiral/roadrunner/` that contains all the necessary source code.

If I don't need the Go runtime, what's the purpose of this dependency?

**Comment by stefanos82** [comment]
I just saw the 'src' folder that contains all the PHP files. Silly me...
